src/error.rs

```diff
@@ -0,0 +1,44 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub struct TokenizersError(pub tokenizers::Error);
+
+impl PartialEq for TokenizersError {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.to_string() == other.0.to_string()
+    }
+}
+
+impl std::fmt::Display for TokenizersError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
```

Comment:
This impl can be done in the struct definition since we're using `thiserror`:
```rust
#[derive(Error, Debug)]
#[error("{0}")] 
pub struct TokenizersError(pub tokenizers::Error);
```